pocs/filterwheel/simulator.py

```diff
@@ -0,0 +1,119 @@
+        move_bidirectional (bool, optional): if True will simulate filter wheel which can rotate in
+            either direction, if False (default) will similate a filter wheel that only moves in
+            one direction.
+    """
+    def __init__(self,
+                 name='Simulated Filter Wheel',
+                 model='simulator',
+                 camera=None,
+                 filter_names=None,
+                 timeout=10 * u.second,
+                 serial_number=None,
+                 move_time=1 * u.second,
+                 move_bidirectional=False,
+                 *args, **kwargs):
+        super().__init__(name=name,
+                         model=model,
+                         camera=camera,
+                         filter_names=filter_names,
+                         timeout=timeout,
+                         serial_number=serial_number,
+                         *args, **kwargs)
+        if isinstance(move_time, u.Quantity):
+            self._move_time = move_time.to(u.second).value
+        else:
+            self._move_time = move_time
+        self._move_birectional = bool(move_bidirectional)
+        self.connect()
+        self.logger.info("Filter wheel {} initialised".format(self))
+
+##################################################################################################
+# Properties
+##################################################################################################
+
+    @AbstractFilterWheel.position.getter
+    def position(self):
+        """ Current integer position of the filter wheel """
+        if math.isnan(self._position):
+            self.logger.warning("Filter wheel position unknown, returning NaN")
+        return self._position
+
+##################################################################################################
+# Methods
+##################################################################################################
+
+    def connect(self):
+        """ Connect to the filter wheel """
+        self._serial_number = 'SW{:04d}'.format(random.randint(0, 9999))
+        self._position = 1
+        self._moving = False
+        self.logger.info("Filter wheel {} initialised".format(self))
+        self._connected = True
+
+##################################################################################################
+# Private methods
+##################################################################################################
+
+    def _move_to(self, position, move_event):
+        if self._moving:
+            move_event.set()
+            msg = "Attempt to move filter wheel when already moving"
+            self.logger.error(msg)
+            raise RuntimeError(msg)
+
+        move_distance = position - self.position
+        if move_distance < 0:
+            if not self._move_birectional:
+                move_distance += self._n_positions
+            else:
+                move_distance = -move_distance
```

Comment:
Can you replace the above with a simple `move_distance = position % self._n_positions`?

Reply:
Sort of...
```
move_distance = position - self.position
if self._move_bidirectional:
    move_distance = abs(move_distance)
else:
    move_distance = move_distance & self._n_positions
```
is equivalent, 1 line shorter, and arguably clearer.